rc/tools/spell.kak

```diff
@@ -87,57 +87,60 @@ define-command spell-clear %{
     unset-option buffer spell_regions
 }
 
-define-command spell-next %{ evaluate-commands %sh{
+define-command spell-rel-jump -hidden -params 0..1 %{ evaluate-commands %sh{
     anchor_line="${kak_selection_desc%%.*}"
     anchor_col="${kak_selection_desc%%,*}"
     anchor_col="${anchor_col##*.}"
 
-    start_first="${kak_opt_spell_regions%%|*}"
-    start_first="${start_first#* }"
+    spell_first="${kak_opt_spell_regions%%|*}"
+    spell_first="${spell_first#* }"
+    spell_last="${kak_opt_spell_regions##* }"
+    spell_last="${spell_last%|*}"
 
     # Make sure properly formatted selection descriptions are in `%opt{spell_regions}`
-    if ! printf %s "${start_first}" | grep -qE '^[0-9]+\.[0-9]+,[0-9]+\.[0-9]+$'; then
+    if ! printf %s "${spell_first}" | grep -qE '^[0-9]+\.[0-9]+,[0-9]+\.[0-9]+$'; then
         exit
     fi
 
-    printf %s "${kak_opt_spell_regions#* }" | awk -v start_first="${start_first}" \
-                                                  -v anchor_line="${anchor_line}" \
-                                                  -v anchor_col="${anchor_col}" '
-        BEGIN {
-            anchor_line = int(anchor_line)
-            anchor_col = int(anchor_col)
-        }
-
-        {
-            for (i = 1; i <= NF; i++) {
-                sel = $i
-                sub(/\|.+$/, "", sel)
+    if [ "$1" = "-rev" ]; then
+        rev=1
+    elif [ -n "$1" ]; then
+        echo "fail -- Unrecognised parameter $(kakquote "$1")"
+    fi
 
-                start_line = sel
-                sub(/\..+$/, "", start_line)
-                start_line = int(start_line)
+    printf %s "${kak_opt_spell_regions#* }" | \
+        awk -v spell_first="${spell_first}" -v spell_last="${spell_last}" \
+            -v anchor_line="${anchor_line}" -v anchor_col="${anchor_col}" \
+            -v rev="${rev}" -v RS=" " -F '[.,|]' '
+            BEGIN {
+                anchor_line = int(anchor_line)
+                anchor_col = int(anchor_col)
+            }
 
-                start_col = sel
-                sub(/,.+$/, "", start_col)
-                sub(/^.+\./, "", start_col)
-                start_col = int(start_col)
+            {
+                # rev still has anchor at start, so we need to check end of spell
+                col_check = (!rev && $2 > anchor_col) || (rev && $4 >= anchor_col)
+                if ($1 >= anchor_line && ($1 != anchor_line || col_check)) {
```

Comment:
I don't understand why we need a special case for the spell-prev case. Shoulnd't we just look at the starting position of the word (for simplicity)?

Reply:
the comment here... kinda sucks, my bad. the intention is that when using `spell-prev`, the "next word" we should search for should "start" at the end of the word, since the search direction is reversed
imagine the following case where the anchor and cursor are reversed when selecting a word:

```
word misspelled1 word misspelled2 word
                      C---------A
```

if we compare the anchor to the start of the word when using `spell-prev`, it'll find `misspelled2` again, since the start of the word is behind the anchor. if we compared the anchor to the end of the word, it'll find `misspelled1` as intended.

the same situation when using `spell-next` is:

```
word misspelled1 word misspelled2 word
     A---------C
```

since it compares the anchor to the start of the word, it'll find `misspelled2` next, as intended

this kind of relates to the above notion of maybe reworking what is actually compared when searching for the next misspelled word.